File: php/82_Remove_Duplicates_from_Sorted_List_II.php
<?php

/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution 
{

    /**
     * @param ListNode $head
     * @return ListNode
     */
    function deleteDuplicates($head) 
    {
        $dummy = new ListNode();
        $dummy->next = $head;
        $prev = $dummy;
        $current = $head;
        
        while ($current != null) {
            while (
                $current->next != null 
                && $current->val == $current->next->val
            ){
                $current = $current->next;
            }
            if ($prev->next == $current) {
                $prev = $prev->next;
            } else {
                $prev->next = $current->next;
            }
            $current = $current->next;
        }

        return $dummy->next;
    }
}

File: go/82_Remove_Duplicates_from_Sorted_List_II.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    dummy := new(ListNode);
    dummy.Next = head;
    prev := dummy;
    current := head;
    for current != nil {
        for current.Next != nil && current.Val == current.Next.Val {
            current = current.Next;
        }
        if prev.Next == current {
            prev = prev.Next;
        } else {
            prev.Next = current.Next;
        }
        current = current.Next;
    }

    return dummy.Next;
}


